.github/workflows/pull-request.yml

```diff
@@ -27,81 +29,81 @@ env:
+        uses: actions/upload-artifact@v2
+        with:
+          name: html
+          path: _build/html
+
+  deploy:
+    needs: build-basic
+    runs-on: ubuntu-20.04
+    if: ${{ github.event_name == 'push' && github.ref == 'refs/heads/publish' }}
+    steps:
+      - name: download HTML
+        uses: actions/download-artifact@v2
+        with:
+          name: html
+          path: html
 
-      # TODO: change the (bad) default value for SOF_DOC_BUILD in
-      # sof-docs/Makefile and remove this command line override
-      - name: build
-        run: |
-          make html VERBOSE=1 SOF_DOC_BUILD=_build_doxy
-          du -shc _build*/*
+      - name: deploy
+        uses: peaceiris/actions-gh-pages@v3
+        with:
+          personal_token: ${{ secrets.ACTIONS_DEPLOY_KEY }}
+          publish_dir: ./html/
+          publish_branch: master
+          external_repository: xiulipan/xiulipan.github.io
 
-  lax:
 
-    name: "PIP_IGNORE_INSTALLED=0 requirements-lax.txt"
+  build-lax:
+
+    name: "build with requirements-lax.txt"
     runs-on: ubuntu-20.04
     # Makefile downgrades the Sphinx warnings, they are not errors any more
     env: {LAX: 1}
     steps:
       - uses: actions/checkout@v2
 
-      - name: apt-get install base dependencies
-        run: sudo apt-get -y install $ubuntu_base_deps
-
-      - name: temporary python3-pil HACK because github is out of date
-        run: sudo apt-get -y install libimagequant0 libwebpdemux2 &&
-           wget http://security.ubuntu.com/ubuntu/ubuntu/pool/main/p/pillow/python3-pil_7.0.0-4ubuntu0.2_amd64.deb &&
-           sudo dpkg -i python3-pil_*.deb
-
-      - name: apt-get instead of pip
-        run: sudo apt-get -y install
-               python3-pip python3-setuptools python3-wheel
-               python3-sphinx python3-breathe python3-docutils
-               python3-sphinx-rtd-theme python3-sphinxcontrib.plantuml
-
-      - name: PATH += .local/bin
-        run: echo "$HOME/.local/bin" >> $GITHUB_PATH
-
-      # should be a no-op
-      - name: 'pip install -r scripts/requirements-lax.txt'
-        run: pip install -r scripts/requirements-lax.txt
-
-      - name: config tweaks
-        run: |
-           # sphinx-build < 1.7 doesn't support "auto" and we're not sure
-           # all default modules are "thread-safe"
-           sed -i -e '/SPHINXBUILD/ s/-j *auto/-j 1/' Makefile
-           # We don't want plantUML to raise the contribution bar
-           sed -i -e 's/^\(plantuml_output_format *=\).*/\1 "none"/' conf.py
-
-      # No YAML anchors; this is copy/paste
-      - name: configure SOF doc
-        run: |
-          git clone https://github.com/thesofproject/sof
-          cmake -GNinja -S sof/doc -B _build_doxy
-
-      - name: build
+      - name: setup env with requirements-lax.txt
         run: |
-          make html VERBOSE=1 SOF_DOC_BUILD=_build_doxy
-          du -shc _build*/*
+          sudo apt-get update
```

Comment:
apt-get update will work forever. This is bad because it's very slow. We want the wget URL to break in the future, this will remind us to remove it and be fast again.

Reply:
I had some test and find apt update will be faster than wget URL workaround. So I decide to use a more simpler way.
WGET way (total 2m48s build takes 57s, setup takes1m51s)
https://github.com/xiulipan/sof-docs/runs/1783208577?check_suite_focus=true
Update way(total 1m38s build takes 54s, setup takes 54s)
https://github.com/xiulipan/sof-docs/runs/1784299347?check_suite_focus=true

Reply:
I'm not comparing the speed of apt-update and wget. They're both slow. I want wget because it will break in the future, then we can remove it be fast. apt-get update will be slow forever. Anyway this is off-topic, totally unrelated to deploying the docs.

Reply:
why it break then will be fast?
I just did some clean up to speed up my test.

Reply:
> why it break then will be fast?

I'm not parsing this sorry. 

> I just did some clean up to speed up my test.

I do this all the time and I still submit unrelated changes in different PRs. Use `git rebase -i` and push HEAD~n

(I doubt merging the steps made any significant speed difference)

Reply:
> I want wget because it will break in the future, then we can remove it and be fast again.
I am just think why will above situation happen?

If you are caring about this, I am totally OK to spilt the PR if you all think this is needed. This only a draft to test all my expected change to achieve the final goal here (move deploy from Travis to GitHub action)

Reply:
Thanks, what I care about is not noising and slowing down this important PR and feature with github UI discussions, optimizations, Ubuntu workarounds and other unrelated disagreements.